core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -115,7 +116,12 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
         Computer c;
         c = byNameMap.get(n.getNodeName());
         if (c!=null) {
-            c.setNode(n); // reuse
+            try {
+                c.setNode(n); // reuse
+                used.add(c);
+            } catch (RuntimeException e) {
+                LOGGER.log(Level.WARNING, "Error updating node " + n.getNodeName() + ", continuing", e);
```

Comment:
Better avoid concatenating, but well, this code path will not be called often and JUL is crappily verbose to do the same the right way :-(